src/components/ApiAction.vue

```diff
@@ -0,0 +1,416 @@
+                    <b-col cols="12" class="text-center my-3 pointer">
+                      <b data-cy="api-actions-tab-plus">+</b>
+                    </b-col>
+                  </b-row>
+                </template>
+              </b-tabs>
+            </b-card>
+          </b-col>
+        </b-row>
+      </b-container>
+    </div>
+    <SaveQueryModal
+      :isQueryNameValid="isQueryNameValid"
+      @storeNewQuery="storeNewQuery"
+    />
+  </Multipane>
+</template>
+
+<script>
+import SaveQueryModal from '@/components/ApiAction/SaveQueryModal'
+import QueryList from '@/components/ApiAction/QueryList'
+import QueryCard from '@/components/ApiAction/QueryCard'
+
+import { Multipane, MultipaneResizer } from 'vue-multipane'
+import { mapGetters } from 'vuex'
+import _ from 'lodash'
+import { truncateName } from '@/utils'
+
+export default {
+  components: {
+    Multipane,
+    MultipaneResizer,
+    QueryCard,
+    SaveQueryModal,
+    QueryList
+  },
+  data() {
+    return {
+      tabs: [],
+      currentTabIdx: 0,
+      showAlert: true,
+      api: null,
+      openapi: null,
+      loading: true,
+      savedQueries: [],
+      newSaveTabIdx: null
+    }
+  },
+  computed: {
+    ...mapGetters('kuzzle', ['$kuzzle', 'currentEnvironment']),
+    ...mapGetters('auth', ['canGetPublicApi', 'canGetOpenApi']),
+    controllers() {
+      return this.api ? Object.keys(this.api) : []
+    },
+    actions() {
+      if (!this.tabs[this.currentTabIdx] || !this.api) {
+        return []
+      }
+      const currentController = this.tabs[this.currentTabIdx].query.controller
+      return currentController && this.api[currentController]
+        ? Object.keys(this.api[currentController])
+        : []
+    },
+
+    emptyTab() {
+      return {
+        query: {
+          controller: null,
+          action: null,
+          body: {}
+        },
+        saved: false,
+        name: '',
+        response: '',
+        savedIdx: null
+      }
+    },
+    currentQueryBody() {
+      if (!this.tabs[this.currentTabIdx]) {
+        return null
+      }
+      return this.tabs[this.currentTabIdx].body
+    },
+    currentQueryName() {
+      if (!this.tabs[this.currentTabIdx]) {
+        return null
+      }
+      return this.tabs[this.currentTabIdx].name
+    }
+  },
+  methods: {
+    closeAlert() {
+      this.showAlert = false
+    },
+    formatTabName(tabContent) {
+      let name = tabContent.name ? tabContent.name : 'New API action'
+      name = truncateName(name, 11)
+      name += tabContent.saved ? '' : ' *'
+      return name
+    },
+    queryChanged({ query, tabIdx }) {
+      const savedIdx = this.tabs[tabIdx].savedIdx
+      if (savedIdx !== null && savedIdx !== undefined) {
+        this.tabs[tabIdx].saved = _.isEqual(
+          this.savedQueries[savedIdx].query,
+          query
+        )
+      }
+      this.tabs[tabIdx].query = JSON.parse(JSON.stringify(query))
```

Comment:
Ok, I feel this is the kind of things we do to work around the limits of the JSONEditor but... why do you do this?

Reply:
even `_.clone ` from lodash keep reference to the initial object 
I had a lot of trouble concerning the json editor and the object references so i decided to add parse/stringify for every json editor interactions

Reply:
Ok, I'd like to talk about this on Discord, I'm very sensitive to the JSONEditor bad DX...